From e60aa444e17cec649a05d9b8040c9393413e5be9 Mon Sep 17 00:00:00 2001
From: Eric Ernst <eric.ernst@intel.com>
Date: Sun, 4 May 2014 11:21:06 -0700
Subject: [PATCH 1/1] Add kernel parameter for kernel version

Create a kernel cmdline parameter, "version_addendum", which can be
used to add text to the kernel version that is reported from
/proc/version.

Change-Id: Ic773dcb14df4de8e612e7de2df46076e50e7f31b
Signed-off-by: Eric Ernst <eric.ernst@intel.com>
---
 fs/proc/version.c |   14 +++++++++++++-
 init/version.c    |    2 +-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/fs/proc/version.c b/fs/proc/version.c
index d2154eb6d78f..1ea545d471a1 100644
--- a/fs/proc/version.c
+++ b/fs/proc/version.c
@@ -5,12 +5,17 @@
 #include <linux/seq_file.h>
 #include <linux/utsname.h>
 
+#define ADDENDUM_LENGTH 20
+static char version_addendum[ADDENDUM_LENGTH];
+
 static int version_proc_show(struct seq_file *m, void *v)
 {
 	seq_printf(m, linux_proc_banner,
 		utsname()->sysname,
 		utsname()->release,
-		utsname()->version);
+		utsname()->version,
+		version_addendum);
+
 	return 0;
 }
 
@@ -26,6 +31,13 @@ static const struct file_operations version_proc_fops = {
 	.release	= single_release,
 };
 
+static int __init set_version_addendum(char *str)
+{
+	strncpy(version_addendum, str, ADDENDUM_LENGTH);
+	return 0;
+}
+early_param("version_addendum", set_version_addendum);
+
 static int __init proc_version_init(void)
 {
 	proc_create("version", 0, NULL, &version_proc_fops);
diff --git a/init/version.c b/init/version.c
index 1a4718e500fe..180059b69b7a 100644
--- a/init/version.c
+++ b/init/version.c
@@ -47,4 +47,4 @@ const char linux_banner[] =
 const char linux_proc_banner[] =
 	"%s version %s"
 	" (" LINUX_COMPILE_BY "@" LINUX_COMPILE_HOST ")"
-	" (" LINUX_COMPILER ") %s\n";
+	" (" LINUX_COMPILER ") %s " "%s\n";
-- 
1.7.9.5

